FWCore/Framework/interface/ESProducts.h

```diff
@@ -47,8 +47,8 @@ namespace edm {
             typedef Null head_type;
          };
 
-         template< typename T> struct OneHolder< std::auto_ptr<T> > {
-            typedef std::auto_ptr<T> Type;
+         template< typename T> struct OneHolder< std::unique_ptr<T> > {
+            typedef std::unique_ptr<T> Type;
             OneHolder() {}
```

Comment:
@Dr15Jones I don't know if auto_ptr is still needed, but to be safe, we'll keep it.